### PR TITLE
Chore: Docblock fixes

### PR DESCRIPTION
**Pull request by brianmuse**
Fixes up docblocks, specifically @param and @return tags. Most of these were discovered by inspection tools in PHPStorm. 

The library is mixed on if it includes `@throws` tags, so I didn't add throws tag everywhere necessary. This PR strictly fixes incorrect types.

### CODE REVIEW

File: lib/Exceptions.php [hunk 1]
```diff
@@ -77,8 +77,8 @@ class UndefinedPropertyException extends ModelException
 	/**
 	 * Sets the exception message to show the undefined property's name.
 	 *
-	 * @param str $property_name name of undefined property
-	 * @return void
+	 * @param string $class_name The name of the class where the property isn't defined.
+	 * @param string $property_name name of undefined property
```
Comment (mcos) [inline lib/Exceptions.php:81]: `The name of the undefined property.`

Comment (mcos) [inline lib/Exceptions.php:81]: Oh wait, for some reason, this isn't actually changed. Nevermind then.

File: lib/Connection.php [hunk 2]
```diff
@@ -160,7 +161,7 @@ private static function load_adapter_class($adapter)
 	 * </code>
 	 *
 	 * @param string $connection_url A connection URL
-	 * @return object the parsed URL as an object.
+	 * @return \stdClass the parsed URL as an object.
```
Comment (jstruzik) [inline lib/Connection.php]: How about just including the class?

File: lib/Model.php [hunk 3]
```diff
@@ -571,7 +572,7 @@ public function attributes()
 	/**
 	 * Retrieve the primary key name.
 	 *
-	 * @param boolean Set to true to return the first value in the pk array only
+	 * @param boolean $first Set to true to return the first value in the pk array only
```
Comment (jstruzik) [inline lib/Model.php:575]: Period.

File: lib/Exceptions.php [hunk 4]
```diff
@@ -103,9 +103,8 @@ class ReadOnlyException extends ModelException
 	/**
 	 * Sets the exception message to show the undefined property's name.
 	 *
-	 * @param str $class_name name of the model that is read only
-	 * @param str $method_name name of method which attempted to modify the model
-	 * @return void
+	 * @param string $class_name name of the model that is read only
```
Comment (jstruzik) [inline lib/Exceptions.php:106]: Periods.

**Comment by jstruzik** [comment]
No fixarino? https://github.com/robinpowered/php-activerecord/blob/master/lib/Model.php#L1834

File: lib/Connection.php (hunk 2 above)
Comment (brianmuse) [inline lib/Connection.php]: What to you mean?

File: lib/Model.php (hunk 3 above)
Comment (brianmuse) [inline lib/Model.php:575]: Intention here isn't to bring this project under our style guide.

Comment (jstruzik) [inline lib/Model.php:575]: Not saying that, but this PR seems pretty inconsistent where periods are added and other places they are not.

File: lib/Connection.php (hunk 2 above)
Comment (jstruzik) [inline lib/Connection.php]: You could just `use stdClass;` at no cost.

**Comment by brianmuse** [comment]
@jstruzik https://github.com/robinpowered/php-activerecord/pull/4/files#diff-8e1db8ebf3425c345973e98193903012R6

File: lib/Connection.php (hunk 2 above)
Comment (brianmuse) [inline lib/Connection.php]: Ooooh yeah :+1:

File: lib/Model.php [hunk 5]
```diff
@@ -3,6 +3,7 @@
  * @package ActiveRecord
  */
 namespace ActiveRecord;
+use Closure;
```
Comment (Rican7) [inline lib/Model.php:6]: I don't see where this is used

Comment (brianmuse) [inline lib/Model.php:6]: Transaction docblock (not shown in diff)

File: lib/Model.php [hunk 6]
```diff
@@ -234,7 +235,7 @@ class Model
 	 * Constructs a model.
 	 *
 	 * When a user instantiates a new object (e.g.: it was not ActiveRecord that instantiated via a find)
-	 * then @var $attributes will be mapped according to the schema's defaults. Otherwise, the given
+	 * then <code>$attributes</code> be mapped according to the schema's defaults. Otherwise, the given
```
Comment (Rican7) [inline lib/Model.php]: if you're going to change this one, you might want to fix the similar references in the same sentence

**Comment by jstruzik** [comment]
tyty

File: lib/Model.php (hunk 6 above)
Comment (brianmuse) [inline lib/Model.php]: Ah yeah. The reason i fixed this one was `@var $attributes will` was considered `{@var $attributes will}` and actually affected `string $attribute  Name of the attribute` in a very odd way. Goal here is to not rewrite the docblocks, just fix them from causing IDE warnings when using the methods.

File: lib/Model.php (hunk 5 above)
Comment (Rican7) [inline lib/Model.php:6]: ah, it's only in the doc-block
you didn't put it in the actual method parameters

Comment (brianmuse) [inline lib/Model.php:6]: No code changes here other than `use` statements.

**Comment by brianmuse** [comment]
Just a reminder from the description:

> This PR strictly fixes incorrect types.

File: lib/Model.php (hunk 6 above)
Comment (Rican7) [inline lib/Model.php]: Yea, it's just awkwardly incomplete to fix it once like this in the sentence and ignore the others. If you're going to change only once instance, I'd suggest just removing the `<code>` bits and leave it consistent with the rest.

Comment (brianmuse) [inline lib/Model.php]: yeah thats fine

File: lib/Model.php (hunk 3 above)
Comment (brianmuse) [inline lib/Model.php:575]: The library itself is inconsistent on this. I didnt modify existing docblocks for grammar or style.

Comment (jstruzik) [inline lib/Model.php:575]: > I didnt modify existing docblocks for grammar or style.

https://github.com/robinpowered/php-activerecord/pull/4/files#diff-c2ede32dd201390a146144c5df4b541cR80

I don't really care though, really would like to just move forward.

Comment (brianmuse) [inline lib/Model.php:575]: That one isn't modified. There was a missing param. But yeah lets move on.

**Comment by Rican7** [comment]
This PR feels kind of awkwardly incomplete, as it's leaving a lot of inconsistencies in places... but its helping a bit with type errors, I guess, so it still works.

**Comment by brianmuse** [comment]
The mission here, to reiterate, is to strictly fix incorrect return/param types in docblocks. Not change code or improve docblocks / style all-around. Trying to make using this library not a nightmare on the IDE while we still work with it.

If anything is incomplete on that front, it should be fixed here.

**Comment by brianmuse** [comment]
This was updated b-t-dubs

**Comment by mcos** [comment]
I'm good with this :+1:

**Comment by jstruzik** [comment]
:+1:

**Pull request merged**